aten/src/ATen/native/vulkan/ops/Padding.cpp

```diff
@@ -0,0 +1,109 @@
+#include <ATen/native/vulkan/ops/Common.h>
+#include <torch/library.h>
+
+namespace at {
+namespace native {
+namespace vulkan {
+namespace ops {
+namespace {
+
+using namespace api::utils;
+
+Tensor reflection_pad2d(const Tensor& self_arg, IntArrayRef padding) {
+  const int pad_dim = padding.size();
+  const IntArrayRef input_size = self_arg.sizes();
+  const int input_dim = input_size.size();
+
+  TORCH_CHECK(
+      pad_dim == 1 || pad_dim == 4,
+      "Padding sizes must be a 1-tuple or 4-tuple!");
+  TORCH_CHECK(input_dim >= 2, "Input tensor must have dim >= 2!");
+
+  api::Context* const context = api::context();
+
+  int pad_left = padding[0];
+  int pad_right = padding[0];
+  int pad_top = padding[0];
+  int pad_bottom = padding[0];
+  if (pad_dim == 4) {
+    pad_right = padding[1];
+    pad_top = padding[2];
+    pad_bottom = padding[3];
+  }
+
+  const Tensor self = self_arg.is_vulkan() ? self_arg : self_arg.vulkan();
+  const vTensor& v_self = convert(self);
+
+  std::vector<int64_t> output_size(input_dim);
```

Comment:
Please use c10::SmallVector<int64_t, 4> (or whatever default size you think is best).  That will save a heap allocation since the first 4 (or the number of your choosing) elements are on the stack.

On the other hand if you want this function to work (or know that this is the only combination that makes sense) only for 4d tensors and below you can use std::array.